pkg/cli/image/manifest/manifest.go

```diff
@@ -282,21 +304,30 @@ func FirstManifest(ctx context.Context, from imagereference.DockerImageReference
 	}
 
 	originalSrcDigest := srcDigest
-	srcManifests, srcManifest, srcDigest, err := ProcessManifestList(ctx, srcDigest, srcManifest, manifests, from, filterFn, false)
+	// Do not return a manifest list, filter here to return appropriate os/arch
+	if o.IsWildcardFilter() {
+		// srcManifest returned to print all os/arch options with error message
+		return srcManifest, ManifestLocation{}, &WildcardFilterErr{}
+	}
+	srcManifests, srcManifest, srcDigest, err := ProcessManifestList(ctx, srcDigest, srcManifest, manifests, from, o.FilterByOS, o.Include, false)
 	if err != nil {
 		return nil, ManifestLocation{}, err
 	}
-	if len(srcManifests) == 0 {
-		return nil, ManifestLocation{}, fmt.Errorf("filtered all images from manifest list")
+	// Should only ever return 1 manifest from here
+	m := len(srcManifests)
+	switch {
```

Comment:
Thinking about this more, I don't see a reason for this just not to truncate the manifest at the first item.  I don't think we care if there are m > 1 manifests, we should just select `srcManifests[0]` if `len(srcManifests) > 0`, return the generic error (I don't think its appropriate to print o.FilterByOS at this layer, but you can return a typed error if you want `ErrNoMatchingImage` that uses the same generic message and then transform it at a higher method level).

I would prefer this not become a method but instead stay a function given this comment and the one above.

Reply:
I see, makes sense. I've updated, FirstManifest back to a function - doing all the error handling higher in append, extract

Reply:
also removed the WildcardFilterErr, found it wasn't necessary  - added a ErrNoMatchingManifestsErr when handling manifest list but no matches found to pass back to append,extract